components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -61,33 +65,52 @@ export default function () {
         }
     }, [team]);
 
-    if (!project) return null;
+    const updateProjectSettings = useCallback(
```

Comment:
Wrapped each of these functions with `useCallback()` to avoid creating new references for each render.